resources/sugarcrm/sugarsession.h

```diff
@@ -69,8 +70,11 @@ class SugarSession : public QObject
     bool readPassword();
     void logout();
 
-private:
+    void setProtocol(SugarProtocolBase *protocol);
+    SugarProtocolBase *protocol() const;
+
     KDSoapGenerated::Sugarsoap *soap();
+private:
```

Comment:
If you make soap() public, please try removing all "friend class" declarations from this class, they are probably not needed anymore.